docs/advanced-examples.md

```diff
@@ -1 +1,344 @@
+
+With a configuration that contains
+
+```yaml
+field: "log4j_file"
+relative-source-path: "src/main/java"
+postfix: ".java"
+```
+
+It will transform an input of
+
+```json
+{
+  "log4j_file": "com.foo.Bar"
+}
+```
+
+into
+
+```json
+{
+  "log4j_file": "src/main/java/com/foo/Bar.java"
+}
+```
+
+For more information and the full set of configuration options, please see the [path extractor documentation](/pipeline/plugins/pathextractor).
+
+### Plugin: Version Tracker
+
+Do you use Git to store your code? Do you have a specific branch deployed to production at all times? If yes, you can use the [version tracker](/pipeline/plugins/versiontracker) plugin to automatically pull the latest commit hash from the code that is currently running in production.
+
+The version tracker will periodically pull the latest version from your git repository and tag all incoming messages with a matching project. As such, may be off for small periods if the version tracker has yet to pull the new commit hash after a successful deploy.
+
+Only git repositories are supported. Authentication is supported over HTTP, HTTPS and SSH. Setting up a read-only SSH key is recommended.
+
+An example configuration of the version tracker looks like this:
+
+```yaml
+projects:
+  mock-logging:
+    repository: https://github.com/nickyu42/mock-elastic-logging
+    branch: refs/heads/master
+    update-interval: 60
+```
+
+This will attach the hash of the latest commit of `master` to all messages whose `project` field is set to `"mock-logging"`.
+
+For more information and the full set of configuration options, please see the [version tracker documentation](/pipeline/plugins/versiontracker).
+
+### Plugin: Load Balancer
+
+The load balancer is a bit of a special plugin. It provides no functionality on its own, but rather provides infrastructure for running multiple workers in parallel. It does this by hosting a "fake" plugin manager, which the worker plugins connect to. As far as the worker plugins are aware, they're just part of a normal pipeline.
+
+For full documentation on the load balancing plugin, please see the [load balancer documentation](/pipeline/plugins/loadbalancer). Please note that the load balancer is a built-in way to support parallelizing your pipeline, but unless you are dealing with an extreme amount of events or have a very slow custom plugin, you will likely not need it. The built-in plugins are able to scale to tens of thousands of messages a second.
+
+### Plugin: Reader
+
+The reader plugin is a debugging plugin intended for testing the behavior of your pipeline and for testing your [custom plugins](/docs/writing-custom-plugin.md). It is not intended for production usage. If you need a plugin that is able to read content from a file or from stdin and publish the contents in the pipeline, consider [writing a custom plugin](/docs/writing-custom-plugin).
```

Comment:
Last link is missing a `.md` at the end:
`[writing a custom plugin](/docs/writing-custom-plugin) -> [writing a custom plugin](/docs/writing-custom-plugin.md)`